src/constructors/constructWithOptions.js

```diff
@@ -11,6 +11,8 @@ export default (css: Function) => {
     }
 
     /* This is callable directly as a template function */
+    // TODO - try modifying this to automatically call extends
+    // and remove all extra wrapper components
```

Comment:
This is actually one of the biggest sources of v8 deopts. If you find a way to move some of these methods over to a class, it might help perf a lot